library/src/main/java/com/google/android/exoplayer/hls/Variant.java

```diff
@@ -27,10 +27,15 @@
   public final String url;
   public final Format format;
 
-  public Variant(int index, String url, int bitrate, String codecs, int width, int height) {
+  public Variant(int index, String name, String url, int bitrate, String codecs, int width, int height) {
     this.url = url;
-    format = new Format(Integer.toString(index), MimeTypes.APPLICATION_M3U8, width, height, -1, -1,
-        -1, bitrate, null, codecs);
+    if (name != null) {
```

Comment:
I don't really feel like mixing up NAME labels and track identifiers, maybe we should store the trackId (or the NAME label) in another variable? What do you think about that?